src/Microsoft.AspNet.Mvc.TagHelpers/Internal/GlobbingUtility.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.Cache.Memory;
+using Microsoft.Framework.FileSystemGlobbing;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    /// <summary>
+    /// Utility methods for <see cref="ITagHelper"/>'s that support attributes containing file globbing patterns.
+    /// </summary>
+    public class GlobbingUtility
+    {
+        private FileProviderGlobbingDirectory _baseGlobbingDirectory;
+
+        public GlobbingUtility(IMemoryCache cache, IFileProvider fileProvider, PathString requestPathBase)
+        {
+            Cache = cache;
+            FileProvider = fileProvider;
+            RequestPathBase = requestPathBase;
+            _baseGlobbingDirectory = new FileProviderGlobbingDirectory(fileProvider, fileInfo: null, parent: null);
+        }
+
+        /// <summary>
+        /// The <see cref="IMemoryCache"/> to cache globbing results in.
+        /// </summary>
+        public IMemoryCache Cache { get; set; }
+
+        /// <summary>
+        /// The <see cref="IFileProvider"/> used to watch for changes to file globbing results.
+        /// </summary>
+        public IFileProvider FileProvider { get; set; }
+
+        /// <summary>
+        /// The base path of the current request (e.g. <see cref="HttpRequest.PathBase"/>).
+        /// </summary>
+        public PathString RequestPathBase { get; set; }
+
+        /// <summary>
+        /// Builds a list of URLs.
+        /// </summary>
+        /// <param name="staticUrl">The statically declared URL. This will always be added to the result.</param>
+        /// <param name="includePattern">The file globbing include pattern.</param>
+        /// <param name="excludePattern">The file globbing exclude pattern.</param>
+        /// <returns>The list of URLs</returns>
+        public virtual IEnumerable<string> BuildUrlList(string staticUrl, string includePattern, string excludePattern)
+        {
+            var urls = new HashSet<string>(StringComparer.Ordinal);
+
+            // Add the statically declared url if present
+            if (staticUrl != null)
+            {
+                urls.Add(staticUrl);
+            }
+
+            // Add urls that match the globbing patterns specified
+            var matchedUrls = ExpandGlobbedUrl(includePattern, excludePattern);
+            foreach (var url in matchedUrls)
+            {
+                urls.Add(url);
+            }
+
+            return urls;
+        }
+
+        private IEnumerable<string> ExpandGlobbedUrl(string include, string exclude = null)
+        {
+            if (string.IsNullOrEmpty(include))
+            {
+                return Enumerable.Empty<string>();
+            }
+
+            var includePatterns = include.Split(',');
```

Comment:
This will provide empty pattern if the user has two `,,` and the test below for `includedPatterns` will be misleading. Use the overload that takes `StringSplitOptions` https://msdn.microsoft.com/en-us/library/ms131448(v=vs.110).aspx